File: python/자료구조/주식가격.py

""" Level 2. 주식가격 """

from collections import deque

def solution(prices):

    answer = []
    queue = deque(prices)

    while queue:

        price = queue.popleft()

        sum = 0
        for p in queue:
            sum += 1
            if p < price:break

        answer.append(sum)

    return answer


print(solution([1, 2, 3, 2, 3]))        # [4, 3, 1, 1, 0]

File: Java/src/자료구조/주식가격.java
package 자료구조;

import java.util.*;

public class 주식가격 {

    public static void main(String[] args) {

        int[] p1 = {1, 2, 3, 2, 3};
        System.out.println(Arrays.toString(solution(p1)));
    }

    public static int[] solution(int[] prices) {

        int length = prices.length;

        int[] answer = new int[length];
        for (int i = 0; i < length; i++) {
            int stan = prices[i];
            int count = i;
            for (int j = i+1; j < length; j++) {
                count += 1;
                if (prices[j] < stan) break;
            }
            answer[i] = count - i;
        }

        return answer;
    }
}
